spec/forms/publish/course_study_mode_form_spec.rb

```diff
@@ -13,23 +13,43 @@ module Publish
 
       describe '#save!' do
         it 'does not call the course.ensure_site_statuses_match_study_mode' do
-          expect(course).to receive(:changed?)
-          expect(course).not_to receive(:ensure_site_statuses_match_study_mode)
-          subject.save!
+          expect(subject.save!).to be false
         end
       end
     end
 
     context 'when params are study mode is part_time' do
-      let(:params) { { study_mode: 'part_time' } }
+      let(:params) { { study_mode: ['part_time'] } }
 
       describe '#save!' do
         it 'does calls the course.ensure_site_statuses_match_study_mode' do
           expect(course).to receive(:changed?)
           expect(course).not_to receive(:ensure_site_statuses_match_study_mode)
           subject.save!
+
+          expect(course.study_mode).to eq 'part_time'
         end
       end
     end
+
+    context 'when params are study mode is both part time and full time' do
+      let(:params) { { study_mode: %w[full_time part_time] } }
+
+      it 'does calls the course.ensure_site_statuses_match_study_mode' do
+        subject.save!
+
+        expect(course.study_mode).to eq 'full_time_or_part_time'
```

Comment:
I might be missing something here, how does checking the value of the `study_mode` relate to the `#ensure_site_statuses_match_study_mode`?

Reply:
I may have an answer to this question from past me https://dcyoung.dev/shorts/london-or-the-chicago-school-of-tdd

Reply:
The test is actually meant to test the 'save' method given certain params. 

And part of the change we've implemented going from radio to check boxes is whether or not selecting two boxes results in what previous happened when we selected one box.

ie, instead of `study_mode: 'full_time_or_part_time'` in the params, we end up `study_mode: ['full_time', 'part_time']`. So we want to return that given those new params, it saves as expected. 

I think you're right that it doesn't relate to the `ensure_site_statuses_match_study_mode` method -- that should be tested separately, explicitly. I'll make the change.